### DISCUSSION TITLE
Update on the development branch

### DISCUSSION THREAD

**Discussion by kaiyux** [NVIDIA/TensorRT-LLM#424] [Announcements]
Hi,

The TensorRT-LLM team is pleased to announce that we are pushing an update to the [development](https://github.com/NVIDIA/TensorRT-LLM/tree/main) branch(also includes the [Triton backend](https://github.com/triton-inference-server/tensorrtllm_backend/tree/main)) this November 17th, 2023.

This update includes:

* Model Support
  - QWEN (external contribution - thank to @Tlntin and @Zhaohb)
  - Replit Code V-1.5 3B (external contribution https://github.com/NVIDIA/TensorRT-LLM/pull/205 - thanks to @bheilbrun)
* Features
  - MPT MQA/GQA support (external contribution https://github.com/NVIDIA/TensorRT-LLM/pull/205 - thanks to @bheilbrun)
  - Falcon INT4 AWQ support
  - Encoder-decoder models
    - BF16 support
    - Multi-GPU support, including tensor parallelism, pipeline parallelism and hybrid parallelism.
    - T5 family support w/ gated activation, including T5/mT5/Flan-T5 (developed in parallel with https://github.com/NVIDIA/TensorRT-LLM/pull/351 - thanks to @mlmonk and @nqbao11)
    - Remove padding support
  - Limited support for LoRA (limited to QKV projection and only exposed in Python runtime for now) in the GPT model. More support will be released in the coming weeks.
  - C++ runtime
    - Support to gather generation-phase logits.
    - Add options to return `logProbs` and `cumLogProbs`
  - CUTLASS 3.x upgrade
  - Experimental Python runtime API
* Bug fixes
  - Tensor names error https://github.com/NVIDIA/TensorRT-LLM/issues/369
  - Bloom weights split issue https://github.com/NVIDIA/TensorRT-LLM/issues/374
  - SmoothQuant multi-GPU failure https://github.com/NVIDIA/TensorRT-LLM/issues/267
  - GenerationSession crash issue https://github.com/NVIDIA/TensorRT-LLM/issues/202
  - Typo when calling PyNVML API https://github.com/NVIDIA/TensorRT-LLM/issues/410
* Memory optimization
  - Add utility to load model weight by shard to reduce peak CPU memory usage.
* Performance
  - Add [reproduction steps](https://github.com/NVIDIA/TensorRT-LLM/blob/main/docs/source/performance.md#reproducing-benchmarked-results) for benchmarked performance results.
* Infra
  - NGC docker version upgraded from 23.08 to 23.10.
* Documentation 
  - Add [memory usage documentation](https://github.com/NVIDIA/TensorRT-LLM/blob/main/docs/source/memory.md).
  - Add [instructions](https://github.com/NVIDIA/TensorRT-LLM/tree/main/examples/llama#long-context-length) to run LLaMA that use Position Interpolation for long sequence lengths(32K)

Thanks,
The TensorRT-LLM Engineering Team

**Comment by mlmonk** [comment]
Thank you for these awesome updates. Does this mean that the enc_dec models now support inflight batching?

**Reply by symphonylyh** [reply]
enc_dec's C++ runtime, Triton backend, and inflgiht batching are on the roadmap, not supported yet. Especially for inflight batching, since it's a multi-engine workflow rather than decoder-only, we need to have more investigation

**Reply by shannonphu** [reply]
@symphonylyh Do you have any idea when Triton backend or inflight batching for enc_dec models can be available?

**Reply by symphonylyh** [reply]
@shannonphu enc-dec's Triton backend will come first likely in this month (i.e., next release). enc-dec's inflight batching is pending discussion and I cannot provide an estimate yet.

**Reply by shannonphu** [reply]
@symphonylyh is there any update on when enc-dec Triton backend will become available? I believe the latest releases/announcements did not mention it. Thanks in advance

**Reply by sihanwang41** [reply]
Hi @symphonylyh , we are also looking for enc_dec inflight batching support, it would be great to have this feature. Thanks!

**Reply by symphonylyh** [reply]
Hi @shannonphu @sihanwang41, yes we're working on it. Right now it's at the stage of adding the C++ runtime. Tentative date for Triton enc-dec support is around mid to late January. Thanks for your patience. btw, let's track the progress under #800
